morpheus/app/views.py

```diff
@@ -632,6 +632,9 @@ async def call(self, request):
             pagination['previous'] = dict(
                 href=f'?from={previous_offset}', pfrom=previous_offset, text=f'{previous_offset + 1} - {max(offset, 0)}'
             )
+        if request.query.get('q') is not None and pagination != {}:
+            for direction, sub_dict in pagination.items():
+                sub_dict['href'] += f'&q={request.query.get("q")}'
```

Comment:
So we're declaring `href` in one `if` statement, then changing it 2 lines later?

Why not something like:

```py
url_params = {}
if q := request.query.get('q') and pagination != {}:
    url_params['q'] = q
if len(hits) == size:
    next_offset = offset + size
    url_params['from'] = next_offset
    pagination['next'] = dict(
        href='?' + urlencode(url_params),
        ...
```
etc